lxd/instance/drivers/driver_lxc.go

```diff
@@ -6627,8 +6662,34 @@ func (d *lxc) FileSFTPConn() (net.Conn, error) {
 		_ = os.Remove(forkfilePath)
 	})
 
-	// Spawn forkfile in a Go routine.
+	// Check if there is an existing SFTP process for this running forkfile.
+	sftpMu.Lock()
+	p, found := sftpProcesses[d.forkfileRunningLockName()]
+	if found {
+		// check that the process underlying conn is not nil
+		if p.conn == nil {
+			return nil, fmt.Errorf("SFTP process is not running yet")
+		}
+
+		return p.conn, nil
+	}
+
+	sftpMu.Unlock()
```

Comment:
Why do we unlock here only to lock again on 6686? This allows the potential for multiple SFTP services to be created.

Reply:
Ah no, you've got the `spawnUnlock` above, so this is fine.

Reply:
thanks, fixing that